auth/src/commonMain/kotlin/com/liftric/auth/core/Exceptions.kt

```diff
@@ -0,0 +1,22 @@
+package com.liftric.auth.core
+
+import io.ktor.http.HttpStatusCode
+
+sealed class IdentityProviderException(val status: HttpStatusCode, message: String): Exception(message) {
+    class CodeMismatch(status: HttpStatusCode, message: String): IdentityProviderException(status, message)
+    class ExpiredCode(status: HttpStatusCode, message: String): IdentityProviderException(status, message)
+    class InternalError(status: HttpStatusCode, message: String): IdentityProviderException(status, message)
+    class InvalidLambdaResponse(status: HttpStatusCode, message: String): IdentityProviderException(status, message)
+    class InvalidParameter(status: HttpStatusCode, message: String): IdentityProviderException(status, message)
+    class InvalidPassword(status: HttpStatusCode, message: String): IdentityProviderException(status, message)
+    class LimitExceeded(status: HttpStatusCode, message: String): IdentityProviderException(status, message)
+    class NotAuthorized(status: HttpStatusCode, message: String): IdentityProviderException(status, message)
+    class ResourceNotFound(status: HttpStatusCode, message: String): IdentityProviderException(status, message)
+    class TooManyFailedAttempts(status: HttpStatusCode, message: String): IdentityProviderException(status, message)
+    class TooManyRequests(status: HttpStatusCode, message: String): IdentityProviderException(status, message)
+    class UnexpectedLambda(status: HttpStatusCode, message: String): IdentityProviderException(status, message)
+    class UserLambdaValidation(status: HttpStatusCode, message: String): IdentityProviderException(status, message)
+    class UserNotConfirmed(status: HttpStatusCode, message: String): IdentityProviderException(status, message)
+    class UserNotFound(status: HttpStatusCode, message: String): IdentityProviderException(status, message)
+    class Unknown(status: HttpStatusCode, val type: String, message: String): IdentityProviderException(status, message)
```

Comment:
Maybe `Undefined ` instead of `Unknown`. I'm open for suggestions 🙂

Reply:
Unknown is usually used for this, so I'd stick with Unknown :)
Well, maybe UnknownError in this case.

Also, in the java/kotlin world all exception classes usually have the `Exception` suffix, so it's obvious you dealing with an exception not with a POJO or other class.

Reply:
Since it is a sealed class it would be IdentityProviderException.Unknown.

Reply:
I could also use a class here and append `Exception` to the subclass name.

Reply:
Ah, didn't see the sealed parent, nvmd, `IdentityProviderException.Unknown` is perfect :)